### PR TITLE
server: define logs RPCs in terms of protos

### PR DESCRIPTION
**Pull request by tamird**
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/8441)

### CODE REVIEW

**Comment by petermattis** [comment]
There is no urgency to #8414. Part of the reason for the low priority is so that it wouldn't consume anyone's time, either the author of the code or the reviewer. I'll try and find some time for a review this afternoon.

---

Reviewed 1 of 1 files at r1.
Review status: 0 of 15 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[util/log/log.proto, line 36 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8441#-KOp1fcTcUnzU53U3GSF:-KOp1fcTcUnzU53U3GSG:bij8mov) ([raw file](https://github.com/cockroachdb/cockroach/blob/911a034670ebefb5643aad5f834f40411688f72b/util/log/log.proto#L36)):*
> ```protobuf
> // a log file.
> message FileDetails {
> 	string program = 1;
> ```

Looks like you have tabs instead of spaces on some of these lines.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8441#-:-KOp2-xhi9AKzvsGCjFp:b-y98exf)*

**Comment by tamird** [comment]
Review status: 0 of 15 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[util/log/log.proto, line 36 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8441#-KOp1fcTcUnzU53U3GSF:-KOp36AmHlCke0NTkd7N:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/911a034670ebefb5643aad5f834f40411688f72b/util/log/log.proto#L36)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary>
> Looks like you have tabs instead of spaces on some of these lines.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8441)*

**Comment by cuongdo** [comment]
:lgtm: w/ question

---

Reviewed 2 of 14 files at r3, 13 of 15 files at r4.
Review status: 13 of 15 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[util/log/clog.go, line 417 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8441#-KOujFDkRum4_53J2ukL:-KOuhjc2N-gmhFELfcBP:b-z2zjfe) ([raw file](https://github.com/cockroachdb/cockroach/blob/a2bb3c6775aeae6cd396ca19336da76670a8c266/util/log/clog.go#L417)):*
> ```Go
> 		entry.Time = t.UnixNano()
> 		entry.File = string(m[3])
> 		line, err := strconv.ParseInt(string(m[4]), 10, 32)
> ```

where does `32` come from? the `Entry` message defines `line` as `int64`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8441#-:-KOujRUvVhK4g_sNgQYh:b-xr419d)*

**Comment by tamird** [comment]
Review status: 13 of 15 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[util/log/clog.go, line 417 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8441#-KOujFDkRum4_53J2ukL:-KOukKvX-1PNbDahKM2A:b-26ejmn) ([raw file](https://github.com/cockroachdb/cockroach/blob/a2bb3c6775aeae6cd396ca19336da76670a8c266/util/log/clog.go#L417)):*
<details><summary><i>Previously, cuongdo (Cuong Do) wrote…</i></summary>
> where does `32` come from? the `Entry` message defines `line` as `int64`
</details>
Good eye, fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8441)*

**Comment by mrtracy** [comment]
:lgtm:

---

Review status: 13 of 15 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8441#-:-KOv061iQWww5HHAXvaH:bnfp4nl)*

File: util/log/log.proto [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+package cockroach.util.log;
+option go_package = "log";
+
+import weak "gogoproto/gogo.proto";
+
+// Entry represents a cockroach structured log entry.
+message Entry {
+  int64 severity = 1 [(gogoproto.casttype) = "Severity"];
+  // Nanoseconds since the epoch.
+  int64 time = 2;
```
Comment (maxlang) [inline util/log/log.proto]: @tamird @mrtracy I'm just starting to use this. Why do we use `Timestamp` in some cases and `Long` in others?

Comment (tamird) [inline util/log/log.proto]: I just ported the existing structure from the log package. Feel free to change this, it should be pretty easy.

**Pull request merged**